runtime/src/main/java/org/corfudb/runtime/view/SequencerView.java

```diff
@@ -130,7 +130,8 @@ public TokenResponse next(UUID... streamIds) {
      * @return address space composed of the trim mark and collection of all addresses belonging to this stream.
      */
     public StreamAddressSpace getStreamAddressSpace(StreamAddressRange streamsAddressesRange) {
-        return getStreamsAddressSpace(Arrays.asList(streamsAddressesRange)).get(streamsAddressesRange.getStreamID());
+        StreamAddressSpace streamAddressSpace = getStreamsAddressSpace(Arrays.asList(streamsAddressesRange)).get(streamsAddressesRange.getStreamID());
+        return streamAddressSpace == null ? new StreamAddressSpace() : streamAddressSpace;
```

Comment:
I think, its better to set it in https://github.com/CorfuDB/CorfuDB/blob/86b7aae3ff1f6025b76cae1a54032d5322e04e83/infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java#L829 because then you can set the trimMark to `Address.NON_EXIST` . Only the sequencer has that information, so it would make sense to construct it in the `SequencerServer` and not the `SequencerView`